.github/workflows/release.yml

```diff
@@ -0,0 +1,139 @@
+        - build: linux-musl
+          os: ubuntu-20.04
+          rust: nightly
+          target: x86_64-unknown-linux-musl
+        - build: linux-gnu
+          os: ubuntu-20.04
+          rust: nightly
+          target: x86_64-unknown-linux-gnu
+
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v3
+
+    - name: Install dependencies
+      shell: bash
+      run: |
+        if [[ "${{ runner.os }}" == "Linux" ]]; then
+          sudo apt-get update && sudo apt-get install -y musl-tools
+        fi
+    
+    - name: Install Rust
+      uses: dtolnay/rust-toolchain@nightly
+      with:
+        target: ${{ matrix.target }}
+
+    - name: Install Rust targets
+      run: rustup target add ${{ matrix.target }}
+      
+    - name: Use Cross
+      shell: bash
+      run: | 
+        cargo install cross -Z sparse-registry
+        if [[ "${{ matrix.os }}" == "macos-13" ]]; then
+          echo "CC=clang" >> $GITHUB_ENV
+        elif [[ "${{ matrix.os }}" == "ubuntu-20.04" ]]; then
+          echo "CC=gcc" >> $GITHUB_ENV
+        fi
+        echo "CARGO=cross" >> $GITHUB_ENV
+        echo "TARGET_FLAGS=--target ${{ matrix.target }}" >> $GITHUB_ENV
+        echo "TARGET_DIR=./target/${{ matrix.target }}" >> $GITHUB_ENV
+    
+
+    - name: Show command used for Cargo
+      run: |
+        echo "cargo command is: ${{ env.CARGO }}"
+        echo "target flag is: ${{ env.TARGET_FLAGS }}"
+        echo "target dir is: ${{ env.TARGET_DIR }}"
+
+    - name: Build release binary
+      run: ${{ env.CARGO }} build --release ${{ env.TARGET_FLAGS }} -Z sparse-registry
+
+    - name: Strip release binary
+      if: matrix.build != 'windows-msvc'
+      run: strip "target/${{ matrix.target }}/release/ezkl"
+    
+    - name: Strip release binary (Windows)
+      if: matrix.build == 'windows-msvc'
+      run: strip "target/${{ matrix.target }}/release/ezkl.exe"
+    
+
+    - name: Build archive
+      if: matrix.build != 'win-msvc'
+      shell: bash
+      run: |
+        mkdir -p build-artifacts
+        tar czvf build-artifacts/ezkl-${{ matrix.build }}.tar.gz -C "target/${{ matrix.target }}/release" ezkl
+        echo "ASSET=build-artifacts/ezkl-${{ matrix.build }}.tar.gz" >> $GITHUB_ENV
+    
+    - name: Build archive (Windows)
+      if: matrix.build == 'win-msvc'
+      shell: bash
+      run: |
+        mkdir -p build-artifacts
+        powershell.exe -Command "Compress-Archive -Path 'target/${{ matrix.target }}/release/ezkl.exe' -DestinationPath 'build-artifacts/ezkl-win.zip'"
+        echo "ASSET=build-artifacts/ezkl-win.zip" >> $GITHUB_ENV    
+        
+    - name: Upload release archive
+      uses: actions/upload-release-asset@v1.0.2
+      env:
+        GITHUB_TOKEN: ${{ secrets.RELEASE_TOKEN}}
+      with:
+        upload_url: ${{ needs.create-release.outputs.upload_url}}
+        asset_path: ${{ env.ASSET }}
+        asset_name: ${{ env.ASSET }}
+        asset_content_type: application/octet-stream
```

Comment:
Add a newline at the end of file to help git diff to differentiate between files